### ISSUE TITLE
RFE: automatically log time against a active ticket

### ISSUE THREAD

**Issue by jumpkick** [gioboa/jira-plugin#2]
I'd love the ability to automatically log time against an active JIRA ticket...   

Ideally, there would be an issue id next to the Project down in the statusbar, when the user clicked on it it would give "Not logging" and a list of active tickets assigned to the user in the selected project.   Then when the user clicked on an issue, it would log time against that ticket when VS Code is not idle and the user spent more than 5 minutes or something.  When the user didn't want to log time, they could select "Not Logging".

**Comment by gioboa** [comment]
Hi @jumpkick,
in the last release 0.3.0 I moved all the jira operations in explorer tab and I'm thinking which is the best workflow for switch "Logging/Not Logging", maybe an action in the explorer?
I understand your request and I have to verify if I can detect when VS Code is idle or not.
In statusbar do you suggest the list only with issues "In Progress" for active user?

Thanks :smile:

**Comment by jumpkick** [comment]
Hmm, I suppose you're talking about a global setting for whether to log time at all, yeah explorer would be ok for that.  Tap it and it could pop up a dialog in the bottom right corner "Jira plugin can log time against Jira tickets, once enabled pick the active ticket assigned to you next to active project in the status bar.  Do you want to enable this?" [Enable, Disable].

Soon as the user enables it, status bar gets a new field next to the Active Project that the user can select something like "Not logging time against anything, select an active issue to start" (something less wordy maybe lol) or they could pick one of the active assigned tickets from the pallette list, then it would show next to the active project.   That way the user doesn't have to completely disable time logging if they are doing something not on a Jira ticket, the can select that first option.

I'm not a UI guy, so feel free to do it a better way... 😄

**Comment by gioboa** [comment]
I'm not UI guy either so for me like this can be a good solution. :+1:

**Comment by jumpkick** [comment]
There is a extension called [timeTrack](https://marketplace.visualstudio.com/items?itemName=shachyjr.timetrack) that does tracking of time when VSCode is active and in focus and stops when you click on a different window...  looking at that code may be of some help with implementing this feature.

Here's the git repo: https://github.com/shachyjr/timetrack

**Comment by jumpkick** [comment]
This snippet suggests that 
https://stackoverflow.com/questions/39569993/vs-code-extension-get-full-path

**Comment by gioboa** [comment]
Cool :+1: 
I saw in Jira docs there is the possibility to store work log in a issue.
[Worlog](https://confluence.atlassian.com/jiracoreserver073/logging-work-on-issues-861257349.html)
I think this is the right way to store time activity infos.

**Comment by jumpkick** [comment]
Yup, that is what I'm referring to...  My boss wants my team to put in how much time we work on each issue in that field, but nobody does it because we have to open up each ticket in Jira and then manually write how long we worked... we are so busy we don't have time.

This would let our boss see somewhat how much time without us needing to each go in and enter it. 😊

**Comment by gioboa** [comment]
Feature in v 0.4.0

**Comment by jumpkick** [comment]
Awesome work!  It worked when I tried it...  I logged 2 minutes to test it out 😄 

Today I only managed to log 45 minutes of work because I was switching between other windows.    I might take a stab at modifying it to stop the timer if I don't come back to a VS for 5 or 10 minutes or stop when the system detects away from keyboard...   But that's really a special case, as it is it works perfectly for long coding sessions.  So thanks again for your awesome work!
